codersdk/notifications.go

```diff
@@ -212,4 +212,72 @@ type UpdateNotificationTemplateMethod struct {
 
 type UpdateUserNotificationPreferences struct {
 	TemplateDisabledMap map[string]bool `json:"template_disabled_map"`
+	PushSubscription    string          `json:"push_subscription,omitempty"`
```

Comment:
If we detach WebPush from notifications entirely, should this land under a separate endpoint too?